rclpy/rclpy/__init__.py

```diff
@@ -242,6 +243,52 @@ def spin(node: 'Node', executor: Optional['Executor'] = None) -> None:
         executor.remove_node(node)
 
 
+def spin_for(node: 'Node', executor: 'Executor' = None, duration_sec: float = None) -> None:
+    """
+    Execute work for some time.
+
+    Callbacks will be executed by the provided executor until the context associated with the
+    executor is shut down or the given time duration passes.
+
+    :param node: A node to add to the executor to check for work.
+    :param executor: The executor to use, or the global executor if ``None``.
+    :param timeout_sec: Seconds to wait (blocking).
+    """
+    executor = get_global_executor() if executor is None else executor
+    try:
+        executor.add_node(node)
+        executor.spin_for(duration_sec)
+    finally:
+        executor.remove_node(node)
+
+
+def spin_until_complete(
+    node: 'Node',
+    condition: Callable[[], bool],
+    executor: Optional['Executor'] = None,
+    timeout_sec: Optional[float] = None,
+) -> None:
```

Comment:
Should this not return something to indicate if the function returned due to the condition being complete or due to a timeout instead?

Reply:
Then that would mean that `spin_until_future_complete` should also something similar?

Reply:
Just so we agree before I do the changes: do we just want a `bool` (`True` means it returned due to the condition being complete, or `False` otherwise), or do we something like `rclcpp::SpinUntilCompleteReturnCode` to indicate `SUCCESS` or `INTERRUPTED` or `TIMEOUT`?